### ISSUE TITLE
Keep images vs. Delete images

### ISSUE THREAD

**Issue by jandubois** [rancher-sandbox/rancher-desktop#3040]
I find it confusing that the "Reset Kubernetes" dialog has a checkbox to "Delete container images", but the "Factory Reset" dialog has a "Keep cached Kubernetes images" checkbox. Both are unchecked, so the dialogs have opposite defaults.

At a minimum I think both dialogs should use the same logic and wording, e.g. use  "Delete" for both checkboxes.

If we really want to have opposing defaults, we could make the box checked by default in the "Factory Reset" dialog.

I understand that in the first case we are throwing away the user images, and in the second case the k3s "system" images, but I think it would be fine to also keep the kubernetes images by default. Thoughts?

**Comment by gunamata** [comment]
I agree. Consistent messaging, behavior helps with the user experience. 

I prefer

In `Reset Kubernetes` ->  `Delete local container images` default to Unchecked
In `Factory Reset` -> `Delete Kubernetes system images` default to Checked

As the `Factory Reset` usually has an inferred meaning of clean sweep, it would be better to not keep the Kubernetes system images by default.

**Comment by jandubois** [comment]
> As the `Factory Reset` usually has an inferred meaning of clean sweep, it would be better to not keep the Kubernetes system images by default.

I disagree with this one. The cached versions are really part of the application, that are downloaded on-demand, and are not bundled with the app because of their size (or because they are released later than the app itself).

The cached files are checksummed, and will be identical if downloaded again.

The reason for a "Factory Reset" is to throw away any custom modifications that a user may have made to the configuration, i.e. the settings, and the VM itself. It allows a fresh start, just as if the app was just installed. Having the cached images already present should not make any difference beyond saving some time and bandwidth.

And you still have the option to delete the images, if you really want to.

I've performed a Factory Reset probably over 200 times, and I don't think I ever wanted to delete the cached images, so having it as the default feels rather wrong to me.

Now, the one time you do want to remove the cached images is when you want to uninstall the app, and have no intention of re-installing. But that is a different use-case than a factory reset, and we should have a proper uninstall tool for that instead, which also removes the app, and not just the data.

**Comment by jandubois** [comment]
This is kind of unrelated, but I can see the need for a cache-pruning tool or mechanism: Over time you will accumulate more and more k8s versions in the cache, and on a space-constrained laptop you may want to reclaim the disk space from versions you no longer use.

But in this case you want to delete individual versions instead of all of them. And more importantly, you want to do this without losing your images and workloads, so a factory reset is not the mechanism for this.

**Comment by jandubois** [comment]
I just checked `rdctl factory-reset` and it has a `--remove-kubernetes-cache` option, which is yet another wording choice, and has the opposite default from the UI dialog.

So I want to rename it to `--delete-kubernetes-cache` to align it with the proposed text for the dialog. And I continue to think that the default in the dialog should also be to **not** delete the cache.
